### PR TITLE
Add more openat() helpers of utility functions

### PR DESCRIPTION
**Pull request by daandemeyer**
Split out of #25836

### CODE REVIEW

**Review by yuwata — APPROVED** [review]
Please add assertions for `dir_fd`s.

File: src/shared/smack-util.c [hunk 1]
```diff
@@ -277,13 +274,13 @@ int mac_smack_copy(const char *dest, const char *src) {
 }
 #endif
 
-int rename_and_apply_smack_floor_label(const char *from, const char *to) {
+int renameat_and_apply_smack_floor_label(int fdf, const char *from, int tfd, const char *to) {
```
Comment (poettering) [inline src/shared/smack-util.c]: fdf? ffd, no?

**Comment by poettering** [comment]
lgtm

**Comment by DaanDeMeyer** [comment]
Fixed comments, setting green label

**Pull request merged**